### ISSUE TITLE
User defined conversions to base should be illegal.

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#12809]
Make sure there is no loophole via tuple names.
I did not try, but from the implementation it seems the following might incorrectly compile.

```C#
public class B : A<(int a, int b)>
{
    // conversion from B to its base should be illegal
    public static explicit operator A<(int, int)>(B x)
    {
        return null;
    }
}
```

**Comment by jcouv** [comment]
Relates to this comment: https://github.com/dotnet/roslyn/pull/12689/files/889b690c29f0f2cffd56b16a7f92a6bf044a0310#r72718049

**Comment by jcouv** [comment]
Was fixed by #12904
